### ISSUE TITLE
Pixelated textures when using S3TC texture compression

### ISSUE THREAD

**Issue by Razish** [JACoders/OpenJK#442]
Only confirmed on Linux x86_64 SP. Have not noticed on other platforms, or MP.

When using `r_ext_compress_textures 1` (default) everything appears very pixelated as if playing with nearest filtering

[compressed](https://dl.dropboxusercontent.com/u/41352148/openjk/2013-11-21-013223.jpg) vs [uncompressed](https://dl.dropboxusercontent.com/u/41352148/openjk/2013-11-21-013705.jpg)

Relevant parts of console:

```
----- Initializing Renderer ----
Trying to load "rdsp-vanilla_x86_64.so" from "."...
-------------------------------
SDL using driver "x11"
Initializing OpenGL extensions
...GL_S3_s3tc available
...GL_EXT_texture_compression_s3tc available
...no tc preference specified
.....using GL_EXT_texture_compression_s3tc

GL_VENDOR: Intel Open Source Technology Center
GL_RENDERER: Mesa DRI Intel(R) Sandybridge Mobile 
GL_VERSION: 3.0 Mesa 9.2.1
texturemode: GL_LINEAR_MIPMAP_LINEAR
picmip: 0
compressed textures: enabled
compressed lightmaps: disabled
texture compression method: GL_EXT_texture_compression_s3tc
```

**Comment by smcv** [comment]
I think this might be caused by the "S2TC" library, which deliberately implements a lower-quality version of S3TC to be able to interoperate with compressed textures while avoiding software patents. See #663, http://lists.freedesktop.org/archives/mesa-dev/2011-August/010039.html and https://wiki.freedesktop.org/dri/S3TC/

**Comment by PeterZhizhin** [comment]
Does it still happen?

**Comment by Razish** [comment]
Yes, using S3TC with mesa drivers still results in pixelated images. S3TC is just disabled by default now for mesa drivers.
